src/vaults/AsyncRequestManager.sol

```diff
@@ -226,64 +238,35 @@ contract AsyncRequestManager is BaseRequestManager, IAsyncRequestManager {
         ShareClassId scId,
         address user,
         AssetId assetId,
-        uint128 assets,
-        uint128 shares
+        uint128 fulfilledAssets,
+        uint128 fulfilledShares,
+        uint128 cancelledShares
     ) public auth {
         IAsyncRedeemVault vault_ = IAsyncRedeemVault(address(vault[poolId][scId][assetId]));
 
         AsyncInvestmentState storage state = investments[vault_][user];
         require(state.pendingRedeemRequest != 0, NoPendingRequest());
 
+        if (cancelledShares > 0) {
+            require(state.pendingCancelRedeemRequest == true, NoPendingRequest());
+            state.claimableCancelRedeemRequest = state.claimableCancelRedeemRequest + cancelledShares;
+        }
+
         // Calculate new weighted average redeem price and update order book values
         state.redeemPrice = _calculatePriceAssetPerShare(
-            vault_, ((maxRedeem(vault_, user)) + shares).toUint128(), state.maxWithdraw + assets, MathLib.Rounding.Down
+            vault_,
+            ((maxRedeem(vault_, user)) + fulfilledShares).toUint128(),
+            state.maxWithdraw + fulfilledAssets,
+            MathLib.Rounding.Down
         );
-        state.maxWithdraw = state.maxWithdraw + assets;
-        state.pendingRedeemRequest = state.pendingRedeemRequest > shares ? state.pendingRedeemRequest - shares : 0;
-
-        if (state.pendingRedeemRequest == 0) delete state.pendingCancelRedeemRequest;
-
-        vault_.onRedeemClaimable(user, assets, shares);
-    }
-
-    /// @inheritdoc IRequestManagerGatewayHandler
-    function fulfillCancelDepositRequest(
-        PoolId poolId,
-        ShareClassId scId,
-        address user,
-        AssetId assetId,
-        uint128 assets,
-        uint128 fulfillment
```

Comment:
See above for my explanation why both parameters are represented by the same value in v3